tools/owl-vision/src/search.ts

```diff
@@ -0,0 +1,255 @@
+			return undefined;
+		}
+
+		const result = await this.find(componentName, `class\\s+(${componentName})\\s+extends`, this.componentCache, "js");
+		if (result) {
+			return result;
+		}
+
+		vscode.window.showWarningMessage(`Could not find a component for "${componentName}"`);
+		return undefined;
+	}
+
+	public async findTemplate(templateName: string): Promise<SearchResult | undefined> {
+		const isComponentName = templateName.match(/^[A-Z][a-zA-Z0-9_]*$/);
+
+		if (isComponentName) {
+			const componentResult = await this.findComponent(templateName);
+			if (!componentResult) {
+				return undefined;
+			} else {
+				const document = await vscode.workspace.openTextDocument(componentResult.uri);
+				const text = document.getText();
+				const foundTemplateName = this.getTemplateNameInJS(text);
+				if (foundTemplateName) {
+					templateName = foundTemplateName;
+				} else {
+					vscode.window.showWarningMessage(`Could not find the template of component "${templateName}"`);
+					return undefined;
+				}
+			}
+		}
+
+		// vscode.window.showInformationMessage(`Searching for template "${templateName}"`);
+		templateName = templateName.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+		const result = await this.find(templateName, `t-name="(${templateName})"`, this.templateCache, "xml");
+		if (result) {
+			return result;
+		}
+
+		vscode.window.showWarningMessage(`No template found for "${templateName}"`);
+		return undefined;
+	}
+
+	private async findComponentFromTemplateName(templateName: string): Promise<SearchResult | undefined> {
+		const result = await this.find(templateName, `template\\s*=\\s*["'](${templateName})["']`, this.componentCache, "js");
+		if (result) {
+			return result;
+		}
+
+		vscode.window.showWarningMessage(`Could not find a component for "${templateName}"`);
+		return undefined;
+	}
+
+	private getTemplateNameInJS(str: string): string | undefined {
+		const matches = new RegExp(`template\\s*=\\s*["']([a-zA-Z0-9_\\-\\.]+)["']`, 'g').exec(str);
+		if (matches) {
+			return matches[1];
+		} else {
+			return undefined;
+		}
+	}
+
+	private getTemplateNameInXML(str: string): string | undefined {
+		const matches = new RegExp(`t-name="([a-zA-Z0-9_\\-\\.]+)"`, 'g').exec(str);
+		if (matches) {
+			return matches[1];
+		} else {
+			return undefined;
+		}
+	}
+
+	private async find(
+		name: string,
+		searchQuery: string,
+		cache: Map<string, SearchResult>,
+		fileType: "js" | "xml",
+	) {
+		if (cache.has(name)) {
+			return cache.get(name);
+		}
+
+		const include = `{${vscode.workspace.getConfiguration().get(`owl-vision.${fileType}.include`)}}`;
+		const exclude = `{${vscode.workspace.getConfiguration().get(`owl-vision.${fileType}.exclude`)}}`;
+		const files = await this.getFiles(name, include, exclude);
+
+		for (const file of files) {
+			let result = await this.findInFile(file, new RegExp(searchQuery, 'g'), 6);
```

Comment:
Could be `const`. Where does the magic 6 come from? Seems like it sort of works because `class ` and `t-name` are both 6 characters, but this causes the cursor to be outside the quotes for t-name and in the incorrect position if there is more than one space after the class declaration (`class     MyComponent`). The cursor is also in a very weird place when using switch/switchBesides. If you build your regexes to use lookbehind, you won't need an offset at all and it will probably work better in all cases